### ISSUE TITLE
parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"

### ISSUE THREAD

**Issue by grillba** [hashicorp/terraform-provider-azurerm#18244]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.8

### AzureRM Provider Version

3.21.1

### Affected Resource(s)/Data Source(s)

azurerm_security_center_workspace

### Terraform Configuration Files

```hcl
data "azurerm_subscription" "current" {}

resource "azurerm_security_center_workspace" "this" {
  scope        = data.azurerm_subscription.current.id
  workspace_id = "/subscriptions/aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa/resourcegroups/rg-01/providers/microsoft.operationalinsights/workspaces/law-management-01"
}
```

### Debug Output/Panic Output

```shell
│ Error: Reading Security Center Log Analytics Workspace ID: parsing "/subscriptions/aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa/resourcegroups/rg-01/providers/microsoft.operationalinsights/workspaces/law-management-01": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
│ 
│   with module.shared.azurerm_security_center_workspace.this,
│   on ../modules/subscription/main.tf line 46, in resource "azurerm_security_center_workspace" "this":
│   46: resource "azurerm_security_center_workspace" "this" {
```

### Expected Behaviour

Resource deploys successfully similar to 3.18.0

### Actual Behaviour

│ Error: Reading Security Center Log Analytics Workspace ID: parsing "/subscriptions/aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa/resourcegroups/rg-01/providers/microsoft.operationalinsights/workspaces/law-management-01": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
│ 
│   with module.shared.azurerm_security_center_workspace.this,
│   on ../modules/subscription/main.tf line 46, in resource "azurerm_security_center_workspace" "this":
│   46: resource "azurerm_security_center_workspace" "this" {

### Steps to Reproduce

Upgraded from 3.18.0 to 3.21.1 and the deployment of  azurerm_security_center_workspace is now requiring title case for "resourceGroups" in the workspace_id. 

Our configuration has the resource ID in lower case.

**Comment by liuwuliuyun** [comment]
Cause of this issue:

In https://github.com/hashicorp/terraform-provider-azurerm/pull/18098 , the validation function changed from laparse.LogAnalyticsWorkspaceID(workspaceID.(string)) to workspaces.ParseWorkspaceID(*v). In the new parsing function, it use case sensitive by default. So any case insensitive setting is break and causes this issue.

**Comment by liuwuliuyun** [comment]
We have seen similar issue with this using new hashcorp/go-azure-sdk in https://github.com/hashicorp/terraform-provider-azurerm/issues/18104

**Comment by tombuildsstuff** [comment]
@liuwuliuyun this isn't a bug - previously the validation didn't validate this casing when it should have been - this needs to be resolved by correcting the user input / adding a state migration as required.

**Comment by tombuildsstuff** [comment]
Duplicate of #18235 - looks like we need a state migration here to account for the wrong Resource ID being imported previously

**Comment by J0F3** [comment]
Same for `azurerm_monitor_diagnostic_setting` starting with azurerm version 3.21.

@tombuildsstuff Why is this not a bug? Everything works with azurerm version 3.20. Now with 3.21 existing, previous, working deployments are brocken/blocked because of this error. So this is definitely a bug in azurerm which should to be fixed immediately.

Azure Resource IDs are and were never case sensitive. So why does care the azurerm provider now after a minor update (which should not included breaking changes) suddenly about the casing of Azure Resource IDs!?

**Comment by grillba** [comment]
@tombuildsstuff  any chance this can be reopened as it appears unrelated to #18235 and the fixes in 3.22.0 did not correct the issue.

I've done some further debugging and the lower cased "resourcegroup" in the ID is stored in Azure and is being returned by an API call during the plan stage. 

```
2022-09-19T05:19:42.7186185Z 2022-09-19T15:19:18.875+1000 [DEBUG] provider.terraform-provider-azurerm_v3.23.0_x5: AzureRM Response for https://management.azure.com/subscriptions/eba843e0-aaf6-47a4-a30f-bcb1f90e4c08/providers/Microsoft.Security/workspaceSettings/default?api-version=2017-08-01-preview: 
2022-09-19T05:19:42.7186928Z HTTP/2.0 200 OK
2022-09-19T05:19:42.7187339Z Api-Supported-Versions: 1.0
2022-09-19T05:19:42.7187746Z Cache-Control: no-cache
2022-09-19T05:19:42.7188292Z Content-Type: application/json; charset=utf-8
2022-09-19T05:19:42.7188802Z Date: Mon, 19 Sep 2022 05:19:18 GMT
2022-09-19T05:19:42.7189227Z Expires: -1
2022-09-19T05:19:42.7189598Z Pragma: no-cache
2022-09-19T05:19:42.7189859Z Server: Kestrel
2022-09-19T05:19:42.7190350Z Strict-Transport-Security: max-age=31536000; includeSubDomains
2022-09-19T05:19:42.7190826Z Vary: Accept-Encoding
2022-09-19T05:19:42.7191251Z X-Content-Type-Options: nosniff
2022-09-19T05:19:42.7191805Z X-Ms-Correlation-Request-Id: 9b382cbd-28ed-fca1-715c-c158b3cf5cb2
2022-09-19T05:19:42.7192390Z X-Ms-Ratelimit-Remaining-Subscription-Resource-Requests: 749
2022-09-19T05:19:42.7192963Z X-Ms-Request-Id: 8177da67-12d0-4b36-a6b4-60fbf0c8a5bf
2022-09-19T05:19:42.7193674Z X-Ms-Routing-Request-Id: AUSTRALIASOUTHEAST:20220919T051918Z:84d99cb0-a851-4e5f-8112-4543ac57a392
2022-09-19T05:19:42.7193971Z 
2022-09-19T05:19:42.7195578Z {"id":"/subscriptions/eba843e0-aaf6-47a4-a30f-bcb1f90e4c08/providers/Microsoft.Security/workspaceSettings/default","name":"default","type":"Microsoft.Security/workspaceSettings","properties":{"workspaceId":"/subscriptions/25bb711d-fde4-46b2-a618-3ca494ebe01b/resourcegroups/prg-aus-prci-management-01/providers/microsoft.operationalinsights/workspaces/law-aus-brdev-prci-management-01","scope":"/subscriptions/eba843e0-aaf6-47a4-a30f-bcb1f90e4c08"}}: timestamp=2022-09-19T15:19:18.838+1000
2022-09-19T05:19:42.7197899Z 2022-09-19T15:19:18.880+1000 [TRACE] provider.terraform-provider-azurerm_v3.23.0_x5: Called downstream: @module=sdk.helper_schema tf_provider_addr=provider tf_resource_type=azurerm_security_center_workspace tf_rpc=ReadResource @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:1016 tf_req_id=d9259427-4e17-5c63-8a74-c2099b11f421 timestamp=2022-09-19T15:19:18.838+1000
2022-09-19T05:19:42.7200517Z 2022-09-19T15:19:18.880+1000 [TRACE] provider.terraform-provider-azurerm_v3.23.0_x5: Received downstream response: diagnostic_error_count=1 diagnostic_warning_count=0 tf_req_id=d9259427-4e17-5c63-8a74-c2099b11f421 tf_resource_type=azurerm_security_center_workspace tf_rpc=ReadResource @caller=github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:38 @module=sdk.proto tf_provider_addr=provider tf_req_duration_ms=903 tf_proto_version=5.2 timestamp=2022-09-19T15:19:18.838+1000
2022-09-19T05:19:42.7204160Z 2022-09-19T15:19:18.880+1000 [ERROR] provider.terraform-provider-azurerm_v3.23.0_x5: Response contains error diagnostic: diagnostic_severity=ERROR diagnostic_summary="Reading Security Center Log Analytics Workspace ID: parsing "/subscriptions/25bb711d-fde4-46b2-a618-3ca494ebe01b/resourcegroups/prg-aus-prci-management-01/providers/microsoft.operationalinsights/workspaces/law-aus-brdev-prci-management-01": parsing segment "staticResourceGroups": expected the segment "resourcegroups" to be "resourceGroups"" tf_proto_version=5.2 tf_provider_addr=provider tf_req_id=d9259427-4e17-5c63-8a74-c2099b11f421 @caller=github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/diag/diagnostics.go:56 @module=sdk.proto diagnostic_detail= tf_resource_type=azurerm_security_center_workspace tf_rpc=ReadResource timestamp=2022-09-19T15:19:18.838+1000
2022-09-19T05:19:42.7207305Z 2022-09-19T15:19:18.880+1000 [TRACE] provider.terraform-provider-azurerm_v3.23.0_x5: Served request: tf_proto_version=5.2 tf_rpc=ReadResource @module=sdk.proto tf_provider_addr=provider tf_req_id=d9259427-4e17-5c63-8a74-c2099b11f421 tf_resource_type=azurerm_security_center_workspace @caller=github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:759 timestamp=2022-09-19T15:19:18.838+1000
2022-09-19T05:19:42.7209130Z 2022-09-19T15:19:18.880+1000 [DEBUG] provider.terraform-provider-azurerm_v3.23.0_x5: AzureRM Response for https://management.azure.com/subscriptions/293914b3-1a2e-4484-998d-c20169a486a7/providers/Microsoft.Security/workspaceSettings/default?api-version=2017-08-01-preview: 
```

There appears to be no work around other that destroying the resource outside of Terraform or deleting the current resource from state.

Happy to send on the full diagnostics trace if needed.
